### PR TITLE
itest: switch over to routerrpc.SendPaymentV2

### PR DESCRIPTION
**Pull request by joostjager**

### CODE REVIEW

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go [hunk 1]
```diff
@@ -61,13 +62,14 @@ func testMultiHopHtlcLocalChainClaim(net *lntest.NetworkHarness, t *harnessTest,
...
 	defer cancel()
 
-	alicePayStream, err := alice.SendPayment(ctx)
-	if err != nil {
-		t.Fatalf("unable to create payment stream for alice: %v", err)
-	}
-	err = alicePayStream.Send(&lnrpc.SendRequest{
-		PaymentRequest: carolInvoice.PaymentRequest,
-	})
+	_, err = alice.RouterClient.SendPayment(
```
Comment (wpaulino) [inline lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go:65]: Should be V2?

File: lntest/itest/lnd_multi-hop-error-propagation.go [hunk 2]
```diff
@@ -382,19 +390,28 @@ out:
...
 		t.Fatalf("unable to send payment: %v", err)
 	}
-
-	if resp.PaymentError == "" {
-		t.Fatalf("payment should have failed")
+	resp, err = getPaymentResult(stream)
+	if err != nil {
+		t.Fatalf("cannot get payment result: %v", err)
 	}
+	if resp.Status != lnrpc.Payment_FAILED ||
```
Comment (wpaulino) [inline lntest/itest/lnd_multi-hop-error-propagation.go:409]: We could refactor `sendAndAssertSuccess` to be a bit more generic and take the expected response status and reason.

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go (hunk 1 above)
Comment (joostjager) [reply lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go:65]: fixed

**Comment by joostjager** [comment]
@wpaulino notes there are more that don't use V2

File: lntest/itest/lnd_multi-hop-error-propagation.go (hunk 2 above)
Comment (joostjager) [reply lntest/itest/lnd_multi-hop-error-propagation.go]: Create a similar send and assert method for failures

**Comment by joostjager** [comment]
Remaining `routerrpc.SendPayment` calls converted to `routerrpc.SendPaymentV2`

**Pull request merged**